pkg/factory/factory.go

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020 The Kubernetes Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package factory
+
+import (
+	"k8s.io/apimachinery/pkg/api/meta"
+	"k8s.io/kubectl/pkg/cmd/util"
+	"sigs.k8s.io/cli-utils/pkg/inventory"
+)
+
+// Factory is an interface which wraps the kubectl factory and
+// the inventory client.
```

Comment:
In java guice land the name used for an uber-factory or factory-factory used in DI is _provider_.

Might want to call this package `provider`, and this struct `Provider` or `DepProvider`.
Avoids the collision with `sigs.k8s.io/cli-utils/pkg/util/factory`

Reply:
Renamed factory to provider for package and interface. Please have a look.